core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -197,11 +195,11 @@ class UnitActions {
 
 
         if (unit.name == "Great Scientist" && !unit.isEmbarked()) {
-            actionList += UnitAction("Discover Technology", unit.currentMovement >0
+            actionList += UnitAction("Hurry Research", unit.civInfo.tech.currentTechnologyName() != null
```

Comment:
in original game, when currentTechnologyName()=null, the science great scientist get will add to overflow science directly, i think it is a bug.